spring-kafka/src/test/java/org/springframework/kafka/core/DefaultShareConsumerFactoryTests.java

```diff
@@ -197,4 +185,102 @@ void integrationTestDefaultShareConsumerFactory(EmbeddedKafkaBroker broker) thro
 		consumer.close();
 	}
 
+	@Test
+	void integrationTestSharedConsumersDistribution(EmbeddedKafkaBroker broker) throws Exception {
+		final String topic = "embedded-share-distribution-test";
+		final String groupId = "distributionTestGroup";
+		int recordCount = 8;
+		List<String> consumerIds = List.of("client-dist-1", "client-dist-2");
+		Map<String, Set<String>> consumerRecords = runSharedConsumerTest(topic, groupId, consumerIds,
```

Comment:
Right. So, if we talk about all the data received (exclusively) independently of the number of consumers (queue semantics), then I wonder if just shared `List` to be fulfilled by those consumers would be enough for our test.
Then we just verify content of that list and check if there are no duplicates.
At the same time it might be hard to determine that those records indeed come from different consumers.
Not sure how to be, but `Set` is misleading here since it will eliminate duplicates.